inlang/source-code/versioned-interfaces/message/src/interface.ts

```diff
@@ -55,13 +77,44 @@ export const Variant = Type.Object({
 	pattern: Pattern,
 })
 
+export type InputDeclaration = Static<typeof InputDeclaration>
+export const InputDeclaration = Type.Object({
+	type: Type.Literal("input"),
+	name: Type.String(),
+
+	//TODO make this generic so that only Variable-Ref Expressions are allowed
+	value: Expression,
+})
+
+export type LocalDeclaration = Static<typeof InputDeclaration>
+export const LocalDeclaration = Type.Object({
+	type: Type.Literal("local"),
+	name: Type.String(),
+	value: Expression,
+})
+
+export type Declaration = Static<typeof Declaration>
+export const Declaration = Type.Union([LocalDeclaration, InputDeclaration])
+
+export type Translation = Static<typeof Translation>
+export const Translation = Type.Object({
+	languageTag: LanguageTag,
+	declarations: Type.Array(Declaration),
+	/**
+	 * The order in which the selectors are placed determines the precedence of patterns.
+	 */
+	selectors: Type.Array(Expression),
+	variants: Type.Array(Variant),
+})
+
 export type Message = Static<typeof Message>
 export const Message = Type.Object({
```

Comment:
Lets discuss the pros / cons of a separate type for the internal representation. We can prepare the discussion tomorrow.

Reply:
A new type makes sense -next steps tbd with @jldec